pyching.1
.TH "PYCHING" "1" "27 Feb 2006" "" ""
.SH "NAME"
pyching \- cast and interpret I Ching hexagrams
.SH "SYNOPSIS"
.B pyching
.BR \-h | \-\-help
.br 
.B pyching
.BR \-v | \-\-version
.br 
.B pyching
.RB [ \-d | \-\-disable\-version\-check ]
.SH "DESCRIPTION"
.B pyChing
is a Python program that allows you to consult the I Ching.  The I
Ching is an ancient Chinese book of wisdom, which, apart from being
read as a book, has also traditionally been consulted as an oracle.
.B pyChing
allows you to perform an I Ching reading using the coin method, and
then look up a brief interpretation from the I Ching.
.PP 
For further information, consult the online documentation available by
way of the program's
.B Help
menu.
.SH "OPTIONS"
.TP 
.B \-h, \-\-help
Display a summary of command\-line options.
.TP 
.B \-v, \-\-version
Display version information.
.TP 
.B \-d, \-\-disable\-version\-check
Suppress the initial check for appropriate versions of Python and Tk.
.SH "AUTHOR"
.B pyChing
was written by Stephen M. Gava <elguavas@users.sourceforge.net>.
This manual page is based on the one originally written by Colin Watson <cjwatson@debian.org> for
the Debian GNU system. Thanks Colin.


CHANGES
pyChing - changes
==================

27 Feb 2006
-------------
1.2.2
- module deprecation warning addressed

28 May 2003
-------------
1.2.1
- many improvements and rewrites to hexagram information texts
- new hexagram information browser impemented
- manpage now provided for unix style systems
- sample shell script now provided for unix style systems
- improved hexagram information data handling mechanism
- improved and corrected online help and I Ching introductory text
- pyching.png colour icon now included
- fixed dialog button resizing bug
- fixed submodule imports on mac platform
- improved user config directory handling
- improved program module directory handling
 
16 Sep 2000
-------------
1.0.4
- replaced .pyw file missing from .zip archive

15 Sep 2000
-------------
1.0.3
- many minor bug fixes and internal rewrites and tidy-ups

6 Mar 2000
-------------
0.9.65
- help and information display completely rewritten to 
  show formatted hypertext instead of plain text
- chinese ideograms now displayed in hexgram information
- line drawing code additions to compensate for differences
  of behaviour in tk canvas widget on win32 platforms
- much more efficient implementation of program data 
- many small changes to documentation
- miscellaneous code tweaks and tidyups

22 Feb 2000
-------------
0.9.10
- complete rewrite of help and data files, 
based on new sources. Minor changes to help and data
display routines.

14 Feb 2000
-------------
0.9.3
- updated source and documentation to new website and
  contact information.
- source, data and build changes to allow one set of
  release files for all supported platforms.

15 Sep 1999,2000
-------------
0.9.2
- bug in user colour selection code fixed
- a few other minor tidy-ups in the code
- a few minor fixes in the documentation and data files

1 Sep 1999,2000
-------------
0.9.1
- first public release  


pyching.py
##! /usr/bin/env python
#! /usr/bin/python2
##---------------------------------------------------------------------------##
##
## pyChing -- a Python program to cast and interpret I Ching hexagrams
##
## Copyright (C) 1999-2006 Stephen M. Gava
##
## This program is free software; you can redistribute it and/or modify
## it under the terms of the GNU General Public License as published by
## the Free Software Foundation; either version 2 of the License, or
## (at your option) any later version.
##
## This program is distributed in the hope that it will be of some
## interest to somebody, but WITHOUT ANY WARRANTY; without even the 
## implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
## See the GNU General Public License for more details.
##
## You should have received a copy of the GNU General Public License
## along with this program; see the file COPYING or COPYING.txt. If not, 
##  write to the Free Software Foundation, Inc.,
## 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
## The license can also be found at the GNU/FSF website: http://www.gnu.org
##
## Stephen M. Gava
## <elguavas@users.sourceforge.net>
## http://pyching.sourgeforge.net
##
##---------------------------------------------------------------------------##
"""
start-up module for pyching
"""

import sys

#handle command line switches
def CommandLineSwitches():
    if ('-h' in sys.argv) or ('/h' in sys.argv) or('--help' in sys.argv):
        print ' pyChing - command line switches\n'
        print ' -h, --help                   display this help message'
        print ' -v, --version                display pyching version'
        print ' -d, --disable-version-check  disable Python and Tk version check'
        #print ' -c, --console                run the console version of pyChing 
        sys.exit(0)
    elif ('-v' in sys.argv) or ('/v' in sys.argv) or('--version' in sys.argv):
        from pyching_engine import PychingAppDetails 
        pyching = PychingAppDetails(createConfigDir=0)
        print 'pyChing version:',pyching.version
        sys.exit(0)

    if ('-d' in sys.argv) or ('/d' in sys.argv) or('--disable-version-check' in sys.argv):
        sys.stderr.write("warning - Python and Tk version checking disabled!\n")
    else: #do version checking
        import string
        from Tkinter import TkVersion
        if string.split(sys.version)[0] < '1.5.1': #python version check
            sys.stderr.write("Sorry, pyChing requires at least Python 1.5.1\n")
            sys.exit(1)
        if `TkVersion` < '8.0': #tk version check
            sys.stderr.write("Sorry, pyChing requires at least Tk 8.0\n")
            sys.exit(1)

CommandLineSwitches()

#run pyching

#if ('-c' in sys.argv) or ('/c' in sys.argv) or('--console' in sys.argv):
# #run the console version of pyChing - needs tidying up first
# import pyching_interface_console 
#else:

#run pyching Tkinter GUI
import pyching_interface_tkinter

#if we got to here exit cleanly
sys.exit(0)
